# I've tried to do this directly with libnotify, but glib and qt have some conflicts and this is way easier. How do other programs manage?
diff -ur ./wpa_supplicant/wpa_gui-qt4/wpagui.cpp ../wpa_supplicant-2.4-patched/wpa_supplicant/wpa_gui-qt4/wpagui.cpp
--- ./wpa_supplicant/wpa_gui-qt4/wpagui.cpp	2015-03-15 18:30:39.000000000 +0100
+++ ../wpa_supplicant-2.4-patched/wpa_supplicant/wpa_gui-qt4/wpagui.cpp	2015-04-25 01:28:40.613233479 +0200
@@ -10,6 +10,7 @@
 #include <windows.h>
 #endif /* CONFIG_NATIVE_WINDOWS */
 
+#include <stdlib.h>
 #include <cstdio>
 #include <unistd.h>
 #include <QMessageBox>
@@ -1336,13 +1337,10 @@
 void WpaGui::showTrayMessage(QSystemTrayIcon::MessageIcon type, int sec,
 			     const QString & msg)
 {
-	if (!QSystemTrayIcon::supportsMessages())
+	if (isVisible() || quietMode)
 		return;
 
-	if (isVisible() || !tray_icon || !tray_icon->isVisible() || quietMode)
-		return;
-
-	tray_icon->showMessage(qAppName(), msg, type, sec * 1000);
+    system(("notify-send -a " + qAppName() + " -t " + QString::number(sec * 1000) + " '" + msg + "'").toStdString().c_str());
 }
 
 
